XenAdmin/MainWindow.cs

```diff
@@ -2846,18 +2849,60 @@ void SearchPanel_SearchChanged()
         /// </summary>
         private void UpdateHeader()
         {
+            LicenseStatusTitleLabel.Text = string.Empty;
+            LicenseStatusTitleLabel.ForeColor = Program.TitleBarForeColor;
+
             if (navigationPane.currentMode == NavigationPane.NavigationMode.Notifications)
                 return;
 
             if (SearchMode && SearchPage.Search != null)
             {
                 TitleLabel.Text = HelpersGUI.GetLocalizedSearchName(SearchPage.Search);
+                LicenseStatusTitleLabel.Text = string.Empty;
                 TitleIcon.Image = Images.GetImage16For(SearchPage.Search);
             }
             else if (!SearchMode && SelectionManager.Selection.ContainsOneItemOfType<IXenObject>())
             {
                 IXenObject xenObject = SelectionManager.Selection[0].XenObject;
                 TitleLabel.Text = xenObject.NameWithLocation();
+
+                if (xenObject is Pool) 

```

Comment:
I think we should extract this block into a separate method and maybe try to reduce the code repetition